ecr/fetcher.go

```diff
@@ -109,8 +109,19 @@ func (f *ecrFetcher) fetchForeignLayer(ctx context.Context, desc ocispec.Descrip
 	if len(desc.URLs) < 1 {
 		log.G(ctx).Error("cannot pull foreign layer without URL")
 	}
-	// TODO try more than one URL
-	return f.fetchLayerURL(ctx, desc, desc.URLs[0])
+	var (
+		rdc io.ReadCloser
```

Comment:
This doesn't need to exist in the scope outside the for loop, right?  I'd prefer that you declare this inside the for loop, right above where you use it.

```go
var rdc io.ReadCloser
rdc, err = f.fetchLayerURL(ctx, desc, layerURL)
```

Reply:
Yeah, this can be inside the loop. I'll update the change.